### ISSUE TITLE
SNOW-226109: Replace use of deprecated 'requests' package

### ISSUE THREAD

**Issue by smbkr** [snowflakedb/snowflake-connector-nodejs#135]
This package uses the deprecated `requests` package. This has been deprecated since 2020-02-11. Its use is also related to other issues (#134) with this package.

https://github.com/request/request/issues/3142

**Comment by sfc-gh-kwagner** [comment]
@smbkr We have an outstanding ticket for replacing `request` with a more modern HTTP and browser client that supports modern features of NodeJS and is actively maintained. However, doing this is a massive undertaking as we would have to update API calls and potentially rewrite large parts of the NodeJS driver to take advantage of these more performant design patterns. We'll keep you updated as this progresses forward!

**Comment by smbkr** [comment]
Thanks

Would you consider making it easier for open source contributors to help? There's a few things that might help, like putting all the required work in GitHub issues, and making it easier to work on the repo itself (I tried to fix some things in Feb but the documentation for the test runner is lacking, and it seemed like some scripts were missing from the repo)

**Comment by likeshumidity** [comment]
https://github.com/sindresorhus/got/blob/main/documentation/migration-guides.md

**Comment by jogeiss** [comment]
> https://github.com/sindresorhus/got/blob/main/documentation/migration-guides.md
now at https://github.com/sindresorhus/got/tree/main/documentation/migration-guides ??

**Comment by shayas** [comment]
`request` package is now vulnerable.

`yarn audit`:
```
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ moderate      │ json-schema is vulnerable to Prototype Pollution             │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ json-schema                                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=0.4.0                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ snowflake-sdk                                                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ snowflake-sdk > request > http-signature > jsprim >          │
│               │ json-schema                                                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://www.npmjs.com/advisories/1006724                     │
└───────────────┴──────────────────────────────────────────────────────────────┘
```

**Comment by leoromanovsky** [comment]
Since the vulnerability is present ⬆️ folks whose companies have SLAs to our customers to address these will be forced to migrate away from `snowflake-sdk`. Is there a timeline you could share for when `requests` will be removed as a dependency?

**Comment by sfc-gh-santoniuc** [comment]
@leoromanovsky If I am not mistaken, this was supposed to be fixed in the latest 1.6.6 version.
https://github.com/snowflakedb/snowflake-connector-nodejs/releases

**Comment by chrisba11** [comment]
I did a search through the code and wasn't able to find any use of `request`. Is it just a matter of removing it from the package.json?

**Comment by leoromanovsky** [comment]
> @leoromanovsky If I am not mistaken, this was supposed to be fixed in the latest 1.6.6 version. https://github.com/snowflakedb/snowflake-connector-nodejs/releases

Thanks - I've already tried upgrading to latest and the problem still exists.

```
➜ yarn why json-schema
yarn why v1.22.17
[1/4] 🤔  Why do we have the module "json-schema"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "request#json-schema@0.2.3"
info Reasons this module exists
   - "snowflake-sdk#request#http-signature#jsprim" depends on it
   - Hoisted from "snowflake-sdk#request#http-signature#jsprim#json-schema"
info Disk size without dependencies: "256KB"
info Disk size with unique dependencies: "256KB"
info Disk size with transitive dependencies: "256KB"
info Number of shared dependencies: 0
✨  Done in 0.54s.
```

the dep is here https://github.com/snowflakedb/snowflake-connector-nodejs/blob/master/package.json#L35

**Comment by veltman** [comment]
@chrisba11 it doesn't use `request` directly but it uses `requestretry`, which requires `request` as a peerDependency: https://github.com/snowflakedb/snowflake-connector-nodejs/blob/7cfd263c50d6e1e3f16908ea29065c1f33b1a8fc/lib/http/node.js#L51
